src/main/java/org/deepsymmetry/beatlink/DeviceFinder.java

```diff
@@ -180,268 +134,232 @@ public boolean isAddressIgnored(InetAddress address) {
-     * as long as the Virtual CDJ is active.
-     *
-     * @return the devices which have been heard from recently enough to be considered present on the network
-     *
-     * @throws IllegalStateException if the {@code DeviceFinder} is not active
-     */
-    public Set<DeviceAnnouncement> getCurrentDevices() {
-        if (!isRunning()) {
-            throw new IllegalStateException("DeviceFinder is not active");
-        }
-        expireDevices();  // Get rid of anything past its sell-by date.
-        // Make a copy so callers get an immutable snapshot of the current state.
-        return Collections.unmodifiableSet(new HashSet<DeviceAnnouncement>(devices.values()));
+        expireDevices();
     }
 
-    /**
-     * Find and return the device announcement that was most recently received from a device identifying itself
-     * with the specified device number, if any.
-     *
-     * @param deviceNumber the device number of interest
-     *
-     * @return the matching announcement or null if no such device has been heard from
-     *
-     * @throws IllegalStateException if the {@code DeviceFinder} is not active
-     */
-    public DeviceAnnouncement getLatestAnnouncementFrom(int deviceNumber) {
-        ensureRunning();
-        for (DeviceAnnouncement announcement : getCurrentDevices()) {
-            if (announcement.getDeviceNumber() == deviceNumber) {
-                return announcement;
+/**
+ * Discard any knowledge we have about current devices. Called when shutting down, and also by the
+ * {@link VirtualCdj} when it believes the network has changed in a way that makes them no longer
+ * reachable.
+ */
+synchronized void flush() {
+    final Set<DeviceAnnouncement> lastDevices = new HashSet<DeviceAnnouncement>(devices.values());
+    devices.clear();
+    firstDeviceTime.set(0);
+
+    // Report the loss of all our devices, on the proper thread, also outside our lock.
+    SwingUtilities.invokeLater(new Runnable() {
+        @Override
+        public void run() {
+            for (DeviceAnnouncement announcement : lastDevices) {
+                deliverLostAnnouncement(announcement);
             }
         }
-        return null;
+    });
+}
+
+/**
+ * Get the set of DJ Link devices which currently can be seen on the network. These can be passed to
+ * {@link VirtualCdj#getLatestStatusFor(DeviceUpdate)} to find the current detailed status for that device,
+ * as long as the Virtual CDJ is active.
+ *
+ * @return the devices which have been heard from recently enough to be considered present on the network
+ * @throws IllegalStateException if the {@code DeviceFinder} is not active
+ */
+public Map<DeviceReference, DeviceAnnouncement> getSynchronizedDevices() {
+    if (!isRunning()) {
+        throw new IllegalStateException("DeviceFinder is not active");
     }
+    //expireDevices();  // Get rid of anything past its sell-by date.
```

Comment:
Why did you comment this out?

Reply:
Wait, what even is this method? I thought it was `getCurrentDevices()`.

Reply:
Removed.